tests/test_cases/test_external/test_external.py

```diff
@@ -276,3 +276,84 @@ def ztest_ext_exit_error(dut):
     the clean close down of the sim world"""
     yield external(test_ext_function)(dut)
     yield Timer(1000)
+
+
+@cocotb.test()
+def test_external_raised_exception(dut):
+    """ Test that exceptions thrown by @external functions can be caught """
+    # workaround for gh-637
```

Comment:
#637 was closed without visible discussion (I assume it was discussed somewhere else). Can you elaborate a bit more on the workaround and the issue?

Reply:
Nope, from my perspective #637 was closed without justification

Reply:
@stuarthodgson since you closed #637, can you jump in with more insight here please?

Reply:
I think this workaround works by ensuring that there's at least one more task around for the simulator to run, without which #637 would cause it to give up before finishing the main task.

Reply:
I cant seems no reason why that was closed as a valid issue.

Reply:
@imphil: Do you still want elaboration on the comment here, or is confirmation that 637 still applies sufficient for you?

Reply:
Nope, with the clarification from Stu the comment is now perfectly fine.